ciao-controller/controller_test.go

```diff
@@ -1126,6 +1127,50 @@ func TestGetStorage(t *testing.T) {
 	}
 }
 
+func TestStorageConfig(t *testing.T) {
+	var err error
```

Comment:
I'm sooper confused. This patch seems to do so much more than the description says. Should it be rolled into another patch? Or separated?